modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -853,22 +855,29 @@ inline v_reg<_Tp, n> v_muladd(const v_reg<_Tp, n>& a, const v_reg<_Tp, n>& b,
 /** @brief Dot product of elements
 
 Multiply values in two registers and sum adjacent result pairs.
+
+@cond Doxygen_Suppress
+@param ignore_order - if it's true the intrinsic may perform unorder sum between result pairs in some platforms,
```

Comment:
IMO it would be better to have 2 separate functions for these cases. So I prefer to retain `v_dotprod' intrinsic as is and add separate intrinsic e.g. `v_dotprod_fast` for the case of unorder sum. For me it looks like cleaner and less error prone approach.

Reply:
Well the call between your hand but in my opinion, I prefer having boolean parameter over introduce new intrins, it seems more robust to me and I don't see any errors could be caused but wait your choose will make it as a policy we gonna all follow it

Reply:
The reason I prefer to avoid behavior affecting parameter is that it is possible to determine its value at run-time. While in most cases it could be possible for compiler to pre-evaluate the value and optimize-out the condition, there still could be a complex cases that could ruin the performance.
However since we are going to define a policy IMO it make sense to spend some time to figure out all possible advantages and drawbacks for both approaches(and maybe even discover another opportunity) to decide on the option suitable and convenient for as many cases as possible.